packages/android_intent/android/src/main/java/io/flutter/plugins/androidintent/IntentSender.java

```diff
@@ -158,6 +169,45 @@ Intent buildIntent(
         intent.setPackage(null);
       }
     }
+    Boolean needIgnore = ignoredPackages != null && ignoredPackages.size() > 0;
```

Comment:
Can we clean up this code a bit for readability? It's difficult to quickly see the relevant behavior vs the low level execution. Based on what I see, this could probably be reduced to something like:

```
boolean isPackageFilterRequired = ignoredPackages != null && ignoredPackages.size() > 0;

if (isPackageFilterRequired || isChooserDesired) {
  List<Intent> targetIntents = calculateTargetIntents(...);
  
  if (isChooserDesired) {
    String choosePackageToLaunch(targetIntents);
    intent.setPackage(packageToLaunch);
  } else {
    //...The code that would go here has other issues that I'll comment on
    // but this example should give you an idea of the functional decomposition
    // that I'm thinking of.
  }
}

return intent;
```